osbuild.py

```diff
@@ -201,27 +196,51 @@ def print_header(title, options, machine_name):
     print(f"\t# nsenter -a --wd=/root -t `machinectl show {machine_name} -p Leader --value`")
     print()
 
+class Pipeline:
+    def __init__(self, pipeline, content_store):
+        m = hashlib.sha256()
+        m.update(json.dumps(pipeline, sort_keys=True).encode())
+
+        self.id = m.hexdigest()
+        self.base = pipeline.get("base")
+        self.stages = pipeline.get("stages")
```

Comment:
Stages should never be `None`, because then we'd have two ways to specify "no stages". `pipeline.get("stages", [])` sets it to `[]` if it's not given in the input JSON.

Reply:
Good point! Fixed.